jwt-authorizer/templates/README.md

```diff
@@ -0,0 +1,162 @@
+# Deploying
+
+## Prerequisites
+This process assumes your ingress for the hostname has already been set up.
+We do not rely on tls secrets for any deployments under that assumptions
+
+### Getting Client Information
+
+If you do not have a client as secret, you first need to obtain one from CILogon OAuth2 Client ID and secret.
+
+Go here:
+https://cilogon.org/oauth2/register
+
+1. Add Client Name, e.g. "LSST LSP instance SSO"
+2. Contact Email
+3. Add hostname for Home URL
+  - `http://lsst-lsp-instance.example.com`
+4. Add callback URL for oauth2_proxy
+  - `http://lsst-lsp-instance.example.com/oauth2/callback`
+5. This is a private client
+
+6. Select Scopes:
+
+* email
+* profile
+* org.cilogon.userinfo
+
+7. Refresh Token Lifetime - 24 hours
+  - This is not really necessary, we can probably get by without refresh token
+
+Save that information.
+This is your client id and client secret.
+
+### After submission
+
+A separate email is required to CILogonhelp address to apply the client configuration
+from the client `cilogon:/client_id/6ca7b54ac075b65bccb9c885f9ba4a75` to your new
+client.
+
+## Configuring Auth Services
+
+### Prerequisites
+
+- pip, or
+- git, to clone this repo
+- curl, in case you don't have pip or git
+
+#### Source
+Clone this repo, switch to this directory (kube/template).
+
+```
+git clone https://github.com/lsst/jwt_authorizer
+cd jwt_authorizer/kube/template
+```
+
+If you don't have git, you can also get it with curl:
+```
+curl -sSLO https://github.com/lsst/jwt_authorizer/archive/master.tar.gz
+tar xvzf master.tar.gz
+cd jwt-authorizer-master/kube/template
+```
+
+#### Running the init scripts
+init.sh relies on j2cli. By default it will try to use a local version
+then fall back to a version in in docker. If you would like to override that.
+You can override with the J2_BIN for with your own executable. 
+
+You can `pip install j2cli[yaml]` otherwise and it should work too, but it may try to
+install an old version of pyyaml.
+
+If you don't have pip on the system, but at least have python installed, you should be
+able to bootstrap everything, as long as you at least have curl:
+
+```
+curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
+python get-pip.py --user
+pip install --user j2cli[yaml]
+```
+
+You will want to modify your PATH too:
+`PATH=$PATH:$HOME/.local/bin:$HOME/bin`
+
+### Run ./init.sh
+This will gather required input and write out YAML files to a directory for 
+your workspace. Those yaml files must be applied.
```

Comment:
The point of helm is to just start with the values YAML files?  Is this copy pasted from somewhere?  I'm guessing since the next part talks about using kubectl apply, which we shouldn't need to do if we're using helm?

Reply:
I've reworded README.md and moved init.sh to bootstrap.sh. bootstrap.sh is a helper script to help with the generation of secrets/keys and massage them a bit when deploying new instance. I looked into secret generators for helm but I couldn't find anything that was better than this script.

Reply:
The bootstrap script just helps with the secrets generation now.